File: array_c_plus_plus/11.cpp
#include <vector>
#include <algorithm>

using namespace std;

class Solution {
public:
    int maxArea(vector<int>& height) {
        int max_area = 0;
        int i = 0;
        int j = height.size()-1;
        
        while(i < j){
            max_area = max(max_area, min(height[i], height[j]) * (j- i));

            if(height[i] > height[j]){
                j--;
            }else{
                i++;
            }
        }

        return max_area;
    }
};

File: array_python/11.py
# coding=UTF-8
'''
@Author: httermin
@Date: 2019-12-29 10:14:08
'''
import sys
import os

sys.path.append("..")
class Solution(object):
    def maxArea(self, height):
        """
        :type height: List[int]
        :rtype: int
        """
        
        start = 0
        end = len(height) - 1

        max_contrainer = 0

        while start < end:
            new_container = min(height[start], height[end]) * (end - start)
            max_contrainer = max_contrainer if  < max_contrainer else new_container

            if height[start] < height[end]:
                start += 1
            else:
                end -= 1
        return max_contrainer

if __name__ == "__main__":
    solution = Solution()
    